compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1447,6 +1447,13 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             {
                 self.tables.coroutine_kind.set(def_id.index, Some(coroutine_kind))
             }
+            if def_kind == DefKind::Closure
+                && tcx.type_of(def_id).skip_binder().is_coroutine_closure()
```

Comment:
This is kind of gross, to check the `type_of`. I guess we could have `TyCtxt::is_coroutine_closure` or something? idk, probably not that big of a deal, since `type_of` is already being called so it's definitely cached.